index.js

```diff
@@ -61,10 +61,9 @@ function formatAttrs(attributes, opts) {
       output += ' ';
     }
 
-    if (!value && booleanAttributes[key]) {
-      output += key;
-    } else {
-      output += key + '="' + (opts.decodeEntities ? entities.encodeXML(value) : value) + '"';
+    output += key;
+    if ((value !== null && value !== '') || (opts.xmlMode && !booleanAttributes[key])) {
```

Comment:
`booleanAttributes` holds the names of the attributes for which `=""` should never be added. The reason why it's there is to make sure `xmlMode` works exactly the same way as it was working before. It has no effect whatsoever when not in `xmlMode`.